frontend/lib/resolution/can-pass.cpp

```diff
@@ -1052,6 +1052,10 @@ CanPassResult CanPassResult::canPass(Context* context,
                 actualTup->toValueTuple(context) == formalT) {
             return passAsIs();
           }
+          auto got = canConvert(context, actualQT, formalQT);
+          if (got.passes()) {
+            return got;
+          }
```

Comment:
I'm a little nervous here about this logic for TYPE. My reading is that this will allow passing `type (int(32), int(32))` to a formal with intent `type (int(64), int(64))`. I suspect that's not the case; you probably don't want that.

Reply:
I think that's the way to go :)